### PR TITLE
Merge #16949,#15084,#17138,#16689,#15098, #15932

### PR DESCRIPTION
**Pull request by vijaydasmp**

### CODE REVIEW

**Comment by vijaydasmp** [comment]
Some Infra Issue going on
fatal: unsafe repository ('/builds/dashpay/dash' is owned by someone else)
To add an exception for this directory, call:

	git config --global --add safe.directory /builds/dashpay/dash
section_end:1650684202:step_script

**Comment by PastaPastaPasta** [comment]
Hi is this still draft?

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see https://github.com/UdjinM6/dash/commits/pr4792

**Comment by vijaydasmp** [comment]
mac build is failing at rpc/blockchain.cpp:952:30: error: cannot call function 'blockheaderToJSON' while mutex 'cs_main' is held [-Werror,-Wthread-safety-analysis]     arrHeaders.push_back(blockheaderToJSON(::ChainActive().Tip(), pblockindex));
this appears to be an existing code

**Review by UdjinM6 — APPROVED** [review]
ACK

File: src/interfaces/chain.cpp [hunk 1]
```diff
@@ -288,6 +288,11 @@ class ChainImpl : public Chain
     {
         ::mempool.GetTransactionAncestry(txid, ancestors, descendants);
     }
+    void getPackageLimits(unsigned int& limit_ancestor_count, unsigned int& limit_descendant_count) override
```
Comment (PastaPastaPasta) [inline src/interfaces/chain.cpp:291]: nit commit name should be `partial Merge...`

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**